compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -157,6 +157,13 @@ pub enum StackPopCleanup {
     Root { cleanup: bool },
 }
 
+/// Return type of [`InterpCx::pop_stack_frame`].
+pub struct StackPop<'tcx, Prov: Provenance> {
+    pub jump: StackPopJump,
+    pub target: StackPopCleanup,
```

Comment:
So there's a "jump" and a "target"... what does that mean...?

Reply:
`jump` is whatever the caller needs to execute a jump afterwards. `target` is the popped stack frame's `.return_to_block`.

Reply:
Yeah I can read the types myself. :) But those sound like it's twice the same thing (obviously the caller needs to know the target to execute the jump). So there should be a comment here explaining how they interact.

Reply:
I pushed https://github.com/rust-lang/rust/pull/113128/commits/074d58a4d05c21d149248a7de722debf878a9e35 to try to resolve this. I've renamed `StackPopJump` to `ReturnAction` because that seems more correct. I've also switched to the names used in the `Frame`, namely `return_to_block` and `return_place`. Not sure if I _like_ them (`return_to_block` has type `StackPopCleanup`...), but maybe we can have a separate discussion about naming everything.

Either way having "jump", "target" and "destination" fields is confusing even for me.

Reply:
> Either way having "jump", "target" and "destination" fields is confusing even for me.

Yeah, we have a general naming convention where "destination" is the place to write the return value to and "target" the block to jump to upon return. I find these names very far from clear. I am not sure what would be better names though... we should probably discuss that a bit on Zulip.